### PR TITLE
Add TreatDiscoveryWarningsAsErrors setting

### PR DESCRIPTION
**Pull request by johnthcall**
Addresses #1544 by adding a setting TreatDiscoveryWarningsAsErrors which will report discovery warnings as errors to loggers.

### CODE REVIEW

File: src/Adapter/MSTest.TestAdapter/MSTestSettings.cs [hunk 1]
```diff
@@ -107,6 +108,11 @@ public static RunConfigurationSettings RunConfigurationSettings
     /// </summary>
     public bool MapNotRunnableToFailed { get; private set; }
 
+    /// <summary>
+    /// Gets a value indicating whether a not test discovery warnings should be treated as errors.
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/MSTestSettings.cs]:
```suggestion
    /// Gets a value indicating whether or not test discovery warnings should be treated as errors.
```

File: src/Adapter/MSTest.TestAdapter/PublicAPI/PublicAPI.Shipped.txt [hunk 2]
```diff
@@ -85,6 +85,7 @@ Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.MSTestSettings.Paralleliz
 Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.MSTestSettings.TestSettingsFile.get -> string?
 Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.MSTestSettings.TestTimeout.get -> int
 Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.MSTestSettings.TreatClassAndAssemblyCleanupWarningsAsErrors.get -> bool
+Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.MSTestSettings.TreatDiscoveryWarningsAsErrors.get -> bool
```
Comment (Evangelink) [inline src/Adapter/MSTest.TestAdapter/PublicAPI/PublicAPI.Shipped.txt]: Please move this one to the `unshipped` file.

**Review by Evangelink — COMMENTED** [review]
@johnthcall Thanks for the contribution, I have left 2 small remarks and we should be good to merge.

**Comment by johnthcall** [comment]
> @johnthcall Thanks for the contribution, I have left 2 small remarks and we should be good to merge.

Thanks for the review, pushed fixes

**Review by Evangelink — APPROVED** [review]

**Pull request merged**